src/conn/conn_api.c

```diff
@@ -1851,18 +1881,11 @@ __conn_single(WT_SESSION_IMPL *session, const char *cfg[])
         }
         WT_ERR(__wt_file_lock(session, fh, false));
     }
-
     /*
-     * We own the database home, figure out if we're creating it. There are a few files created when
-     * initializing the database home and we could crash in-between any of them, so there's no
-     * simple test. The last thing we do during initialization is rename a turtle file into place,
-     * and there's never a database home after that point without a turtle file. If the turtle file
-     * doesn't exist, it's a create.
+     * Populate WiredTiger file if new connection or WiredTiger file is empty and we are salvaging.
      */
-    WT_ERR(__wt_turtle_exists(session, &exist));
-    conn->is_new = exist ? 0 : 1;
-
-    if (conn->is_new) {
+    WT_ERR(__wt_filesize(session, fh, &size));
+    if (conn->is_new || (cval.val && (size_t)size == 0)) {
```

Comment:
since we are using the config result from gathering "salvage", could you replace `cval.val` to is_salvage? Would be more easier to read and parse.